Src/Plugins/Security/SEC101_004.FacebookAppCredentialsValidator.cs

```diff
@@ -147,14 +147,16 @@ private ValidationState RetrieveInformation<T>(string url, string id, ref string
                     }
 
                     case System.Net.HttpStatusCode.BadRequest:
+                    case System.Net.HttpStatusCode.InternalServerError:
                     {
                         return ReturnUnauthorizedAccess(ref message, asset: id);
                     }
 
                     default:
                     {
-                        message = $"Unexpected response status code: '{httpResponse.StatusCode}'";
-                        return ReturnUnknownAuthorization(ref message, asset: id);
+                        message = CreateUnexpectedResponseCodeMessage(response.StatusCode, asset: id);
```

Comment:
>CreateUnexpectedResponseCodeMessage [](http://example.com/codeflow?start=34&length=35)

TODO. Shouldn't we have a ReturnUnexpectedResponseCodeMessage handler? i.e., is there a reason why we wouldn't standardize the return value here, to ValidationState.Unknown?

Reply:
I like the idea :)
would facilitate in a lot of places!

Reply:
Created the PR:
[https://github.com/microsoft/sarif-pattern-matcher/pull/452](https://github.com/microsoft/sarif-pattern-matcher/pull/452)